### PR TITLE
UCPKN-2282: Implement Session sub-entity.

### PR DESCRIPTION
**Pull request by kp77**
#3 needs to be merged first.

### CODE REVIEW

**Comment by dxvargas** [comment]
The person is being handled is confusing and may result in errors.
For instance if we link a Person to Users and allow to create them on the fly, we have SQL errors like "Column 'name' cannot be null".

When removing a Person, it asks if a user wants to delete this person from the system. It seems confusing because may seem that the target will be removed. But it never removes the target, it just removes the reference sub-entity.

Adding a person always creates a new reference, even if it's pointing to the same target as another existing reference. Not sure it's wrong. But if a reference sub-entity is used only one time, it should always be removed without asking the user... it won't be reused anyway. No?

Since this PR is not specifically about the persons part, it is working even if it should be improved and it has a lot of work that basically fulfills the requirements, I can approve the PR as it is now. I ask for improvements in this part in a follow-up issue.

**Comment by kp77** [comment]
Hi @dxvargas,

Thanks for the feedback!

> For instance if we link a Person to Users and allow to create them on the fly, we have SQL errors like "Column 'name' cannot be null".

This sounds bad, but I couldn't reproduce. How did this happen?

> When removing a Person, it asks if a user wants to delete this person from the system.

I fixed this in [ee9a90d](https://github.com/openeuropa/oe_agenda/pull/4/commits/ee9a90da1db4b1cc991f507995d1c1466e660234).

> Adding a person always creates a new reference...

Yes, currently this is the case, and this is not agreed on yet in the requirements. Let's wait for the second review to see how we should configure it.

**Review by dxvargas — DISMISSED** [review]

**Comment by dxvargas** [comment]
> Hi @dxvargas,
> 
> Thanks for the feedback!
> 
> > For instance if we link a Person to Users and allow to create them on the fly, we have SQL errors like "Column 'name' cannot be null".
> 
> This sounds bad, but I couldn't reproduce. How did this happen?
> 
> > When removing a Person, it asks if a user wants to delete this person from the system.
> 
> I fixed this in [ee9a90d](https://github.com/openeuropa/oe_agenda/pull/4/commits/ee9a90da1db4b1cc991f507995d1c1466e660234).
> 
> > Adding a person always creates a new reference...
> 
> Yes, currently this is the case, and this is not agreed on yet in the requirements. Let's wait for the second review to see how we should configure it.

About the error, these are the steps:

1. Create a new Person with a Reference field referencing an User having the option "Create referenced entities if they don't already exist" checked. The widget of this field should be Autocomplete.
2. Add a Person of the previously created type in a Session in a Day in an Agenda field, by filling the person field with a non existing user (something like XXXXXXXXX will do.
3. Save the content, an SQL error will be visible.

Anyway, I've approved the PR already.

**Comment by kp77** [comment]
Hi @dxvargas,

> Create a new Person with a Reference field referencing an User having the option "Create referenced entities if they don't already exist" checked.

I see, I didn't try to do it with this option. Indeed, I also think this is out of the scope of the agenda module. Not sure if creating users (or any other entity) while creating the person reference entity (on the fly) is supported at this point. As I understand the referenced person entities (i.e. users, nodes etc) are supposed to be already created when the referencing happens in the agenda.

File: tests/src/FunctionalJavascript/SessionEntityTest.php
Comment (brummbar) [inline tests/src/FunctionalJavascript/SessionEntityTest.php]: We can change the label in the field widget. We could use "Moderator" and "Speaker" so it's a bit more clear in the UI.

File: config/install/field.storage.oe_agenda_session.oe_session_details.yml
Comment (brummbar) [inline config/install/field.storage.oe_agenda_session.oe_session_details.yml]: Following discussions in some recent meetings, I think that having a field that doesn't allow markup will be too restrictive in the future. We should use text formatted long instead, and it will be the project that can configure which markup to show.

**Review by brummbar — CHANGES_REQUESTED** [review]

File: tests/src/FunctionalJavascript/SessionEntityTest.php
Comment (kp77) [reply tests/src/FunctionalJavascript/SessionEntityTest.php]: Done in 6b2d3b0

File: config/install/field.storage.oe_agenda_session.oe_session_details.yml
Comment (kp77) [reply config/install/field.storage.oe_agenda_session.oe_session_details.yml]: Done in 0341385

**Review by brummbar — APPROVED** [review]

**Pull request merged**
